tsl/src/bgw_policy/compression_api.c

```diff
@@ -9,51 +9,108 @@
 #include "interval.h"
 #include "license.h"
 #include "utils.h"
+#include "jsonb_utils.h"
 
 /*
  * Default scheduled interval for compress jobs = default chunk length.
  * If this is non-timestamp based hypertable, then default is 1 day
  */
 #define DEFAULT_SCHEDULE_INTERVAL                                                                  \
-	DatumGetIntervalP(DirectFunctionCall7(make_interval,                                           \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(1),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Float8GetDatum(0)))
+	DatumGetIntervalP(DirectFunctionCall3(interval_in, CStringGetDatum("1 day"), InvalidOid, -1))
+
 /* Default max runtime is unlimited for compress chunks */
 #define DEFAULT_MAX_RUNTIME                                                                        \
-	DatumGetIntervalP(DirectFunctionCall7(make_interval,                                           \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Float8GetDatum(0)))
+	DatumGetIntervalP(DirectFunctionCall3(interval_in, CStringGetDatum("0"), InvalidOid, -1))
+
 /* Right now, there is an infinite number of retries for compress_chunks jobs */
 #define DEFAULT_MAX_RETRIES -1
 /* Default retry period for reorder_jobs is currently 1 hour */
 #define DEFAULT_RETRY_PERIOD                                                                       \
-	DatumGetIntervalP(DirectFunctionCall7(make_interval,                                           \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(0),                                        \
-										  Int32GetDatum(1),                                        \
-										  Int32GetDatum(0),                                        \
-										  Float8GetDatum(0)))
+	DatumGetIntervalP(DirectFunctionCall3(interval_in, CStringGetDatum("1 hour"), InvalidOid, -1))
+
+#define POLICY_COMPRESS_PROC_NAME "policy_compression"
+#define CONFIG_KEY_HYPERTABLE_ID "hypertable_id"
+#define CONFIG_KEY_OLDER_THAN "older_than"
+
+int32
+policy_compression_get_hypertable_id(Jsonb *config)
```

Comment:
I guess these functions should ideally be:

```suggestion
policy_compression_get_hypertable_id(const Jsonb *config)
```

to make clear that the json isn't modified.